js/constants/apiRequest.js

```diff
@@ -0,0 +1,13 @@
+export const POST_API = async (apiUrl, data) => {
+  const response = await fetch('https://bootcamp-api.codeit.kr/api/sign-in', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(data),
+  });
+
+  // const result = JSON.parse(await response.text());
+
+  return response;
+};
```

Comment:
매개변수로 받는 apiUrl은 함수 내부에서 사용하고 있지 않네요!

Reply:
많은 범위를 커버할 수 있는 모듈로 만들려면 headers에 넣을 값도 매개변수로 받을 수 있어야 할 것 같아요!

Reply:
매개변수를 객체로 받는 것도 괜찮을 것 같아요!
```js
export const POST_API = async ({ apiUrl, data }) {
```

Reply:
아래와 같은 느낌으로 사용할 수 있도록 모듈이 설계되는 방식도 괜찮을 것 같아요!
- apiRequest.get()
- apiRequest.post()
- apiRequest.put()
- apiRequest.delete()

Reply:
api 모듈에서 응답이 성공인지, 실패인지 확인하고
성공이면 response를 리턴,
실패이면 error를 리턴하는 방식으로 작성하고 
사용하는 쪽에서 try-catch로 처리하는 방식이 좋을 것 같아요!

```suggestion
export const POST_API = async (apiUrl, data) => {
  const response = await fetch('https://bootcamp-api.codeit.kr/api/sign-in', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(data),
  });
  
  const data = await response.json();

  if(response.ok === false) {
    throw new Error(res.error.text);
  }
  
  return data;
};
```